### PR TITLE
feat: visit storage-monitor using clusterIP instead of domain name

### PR DESCRIPTION
**Pull request by tydra-wang**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/metric/monitor_client.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	c.RLock()
+	clusterIP := c.clusterIP
+	c.RUnlock()
+	if clusterIP == "" {
+		return nil, errors.New("clusterIP not initialized")
+	}
+	params := url.Values{"multi-cnfs-nas": []string{pv}}
+	uri := url.URL{
+		Scheme:   "http",
+		Host:     net.JoinHostPort(clusterIP, "11280"),
```
Comment (mowangdk) [inline pkg/metric/monitor_client.go:59]: Move port value to const.

File: pkg/metric/monitor_client.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		Path:     "/metrics",
+		RawQuery: params.Encode(),
+	}
+	resp, err := (&http.Client{Timeout: time.Millisecond * 500}).Get(uri.String())
```
Comment (mowangdk) [inline pkg/metric/monitor_client.go:63]: same as above

**Comment by huww98** [comment]
I would implement this like DNS. Save IP and lastUpdateTime in StorageMonitorClient, and synchronously update the IP if it is too old. This way, we can have a different TTL to "not found" result. Also, we don't pay for this if no one is requesting the metrics.

**Comment by mowangdk** [comment]
> I would implement this like DNS. Save IP and lastUpdateTime in StorageMonitorClient, and synchronously update the IP if it is too old. This way, we can have a different TTL to "not found" result. Also, we don't pay for this if no one is requesting the metrics.

Never mind

**Pull request merged**